resources/default.rb

```diff
@@ -61,15 +61,15 @@
     notifies :run, 'ruby_block[block until operational]', :delayed
   end
 
-  template ::File.join(install_dir, 'bin', 'nexus.vmoptions') do
-    source 'nexus.vmoptions.erb'
-    variables(
-      new_resource.vmoptions_variables.merge(data: new_resource.data)
-    )
-    mode '0644'
+  vmoptions = node['nexus3_criteo']['jvm']['settings'].map do |k, v|
```

Comment:
This is Criteo-specific, please edit with `node['nexus3']['vmoptions_variables']`.